test/jdk/java/lang/management/ThreadMXBean/VirtualThreads.java

```diff
@@ -191,10 +249,13 @@ void testGetThreadCpuTime1() {
      */
     @Test
     void testGetThreadCpuTime2() throws Exception {
-        runInVirtualThread(() -> {
+        ThreadMXBean bean = ManagementFactory.getThreadMXBean();
+        assumeTrue(bean.isThreadCpuTimeSupported(), "Thread CPU time measurement not supported");
+
+        VThreadRunner.run(() -> {
             long tid = Thread.currentThread().threadId();
-            long cpuTime = ManagementFactory.getThreadMXBean().getThreadCpuTime(tid);
-            assertTrue(cpuTime == -1L);
+            long cpuTime = bean.getThreadCpuTime(tid);
+            assertEquals(-1L, cpuTime);
```

Comment:
Am I correct that `getThreadCpuTime(tid)` returns -1 for the current thread is a virtual thread whereas `getCurrentThreadCpuTime` throws UOE in the current implementation?

`getCurrentThreadCpuTime` is specified to be equivalent to calling `getThreadCpuTime(Thread.currentThread().threadId()`.

Reply:
We didn't get this quite right in JDK 19 but I think I've fixed all those issues now.  So assuming the VM supports CPU time for all platform threads, it means:

- isThreadCpuTimeEnabled and isCurrentThreadCpuTimeSupported will return true.
- If getThreadCpuTime(long) is called with the thread ID of a virtual thread then -1 will be returned.
- If getCurrentThreadCpuTime() is called from a virtual thread then -1 will be  returned.

I did another pass over the API docs, update the "Thread CPU time" section, so I hope it is clearer now.